### PR TITLE
feat(diff): add workspace support

### PR DESCRIPTION
**Pull request by wraithgar**
Refactored a bit so that we can more easily change the `top` and
`prefix` params that were being used, and are different under the
workspace context.

Cleaned up the help output while we were in there adding workspace flags.

### CODE REVIEW

File: lib/diff.js [hunk 1]
```diff
@@ -67,86 +64,96 @@ class Diff extends BaseCommand {
...
-  async defaultSpec () {
-    let noPackageJson
-    let pkgName
+  // get the package name from the packument at `path`
+  // throws if no packument is present OR if it does not have `name` attribute
+  async packageName (path) {
+    let name
     try {
-      pkgName = await readPackageName(this.npm.prefix)
+      // TODO this won't work as expected in global mode
```
Comment (ruyadorno) [inline lib/diff.js:107]: I believe this use case is already handled in L114 throwing an EUSAGE error

File: tap-snapshots/test/lib/load-all-commands.js.test.cjs [hunk 2]
```diff
@@ -199,16 +199,14 @@ The registry diff command
...
 Options:
-[--diff <diff> [--diff <diff> ...]] [--diff-name-only]
-[--diff-unified <diff-unified>] [--diff-ignore-all-space] [--diff-no-prefix]
-[--diff-src-prefix <diff-src-prefix>] [--diff-dst-prefix <diff-dst-prefix>]
+[--diff <pkg-name|spec|version> [--diff <pkg-name|spec|version> ...]]
+[--diff-name-only] [--diff-unified <number>] [--diff-ignore-all-space]
+[--diff-no-prefix] [--diff-src-prefix <path>] [--diff-dst-prefix <path>]
 [--diff-text] [-g|--global] [--tag <tag>]
+[-w|--workspace <workspace-name> [-w|--workspace <workspace-name> ...]]
+[-ws|--workspaces]
```
Comment (ruyadorno) [inline tap-snapshots/test/lib/load-all-commands.js.test.cjs:209]: ❤️

**Review by ruyadorno — APPROVED** [review]

File: lib/diff.js (hunk 1 above)
Comment (wraithgar) [reply lib/diff.js:107]: No, read-package-json no longer throws in global mode. That check didn't belong in there.

Comment (ruyadorno) [reply lib/diff.js:107]: oohh I see what you mean 👍

**Pull request merged**